### PR TITLE
added docstrings for cell list

### PR DESCRIPTION
**Pull request by Tobias-Dwyer**
## Description
Add docstrings

## Motivation and Context

there were no docstrings here

Resolves: #???

## Change log

```

```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/hoomd-blue/blob/master/sphinx-doc/credits.rst).

### CODE REVIEW

File: hoomd/md/nlist.py [hunk 1]
```diff
@@ -119,10 +126,13 @@ class Cell(_NList):
 
     Args:
         buffer (float):  Buffer width.
-        exclusions (tuple): ...
+        exclusions (tuple): Set the parameter which excluded pairs from the
+            neighbor list, therefore excluding them from the pair potential
+            calculation.
```
[the remark this answers is not in the document]
Comment (b-butler) [reply hoomd/md/nlist.py:131]: See https://github.com/glotzerlab/hoomd-blue/blob/4324daf075836f229000e42b346fc997cd16754f/hoomd/md/nlist.py#L202 for information about the options available here.

**Comment by b-butler** [comment]
Also, can you use the `imported_members` option in sphinx-docs for building and remove the stencil and tree neighbor lists for now?

File: hoomd/md/nlist.py [hunk 2]
```diff
@@ -369,40 +334,6 @@ def tune_cell_width(self, warmup=200000, min_width=None, max_width=None, jumps=2
 stencil.cur_id = 0
 
 class tree(nlist):
```
Comment (b-butler) [inline hoomd/md/nlist.py:336]: Sorry I didn't want the docstrings deleted but the documentation removed from the sphinx documentation builds.

File: hoomd/md/nlist.py [hunk 3]
```diff
@@ -185,51 +195,6 @@ def detach(self):
 
 
 class stencil(nlist):
```
Comment (b-butler) [inline hoomd/md/nlist.py:197]: Sorry I didn't want the docstrings deleted but the documentation removed from the sphinx documentation builds.

File: hoomd/md/nlist.py [hunk 4]
```diff
@@ -145,9 +154,10 @@ class Cell(_NList):
 
     Examples::
 
-        nl_c = nlist.cell(rebuild_check_delay = 1)
-        nl_c.buffer = 0.5
-        nl_c.exclusions = tuple();
+        nl_c = nlist.Cell(diameter_shift=True,buffer=0.5)
```
Comment (b-butler) [inline hoomd/md/nlist.py:157]:
```suggestion
        nl_c = nlist.Cell(diameter_shift=True, nbuffer=0.5)
```

File: hoomd/md/nlist.py [hunk 5]
```diff
@@ -53,13 +53,20 @@
 
 
 class nlist:
+    R""" 
+
+    Generates a class for other classes that have not been ported.
```
Comment (b-butler) [inline hoomd/md/nlist.py:58]: This class just serves as a parent class for v2 two neighbor list objects.

File: hoomd/md/nlist.py [hunk 6]
```diff
@@ -53,13 +53,20 @@
...
     pass
 
+# To Do: Migrate all hoomdv2 codes still using nlist to _NList
 
 class _NList(_Operation):
     R""" Base class neighbor list.
 
     Methods provided by this base class are available to all subclasses.
+
+
```
Comment (b-butler) [inline hoomd/md/nlist.py:69]: These lines can be removed.

**Review by b-butler — CHANGES_REQUESTED** [review]
It looks like we had some miscommunication on deleting docstrings.

File: hoomd/md/nlist.py (hunk 2 above)
Comment (Tobias-Dwyer) [reply hoomd/md/nlist.py:371]: Added back the deleted docstrings. And removed from the sphinx documentation.

File: sphinx-doc/module-md-nlist.rst [hunk 7]
```diff
@@ -9,12 +9,10 @@ md.nlist
...
 
     md.nlist.Cell
-    md.nlist.stencil
-    md.nlist.tree
 
 .. rubric:: Details
 
 .. automodule:: hoomd.md.nlist
     :synopsis: Neighbor list acceleration structures.
     :members:
```
Comment (joaander) [inline sphinx-doc/module-md-nlist.rst:17]:
```suggestion
    :members: _Nlist, Cell
```
This makes the `_Nlist` documentation available to users.

File: sphinx-doc/module-md-nlist.rst [hunk 8]
```diff
@@ -9,12 +9,10 @@ md.nlist
...
-    :no-inherited-members:
+    :inherited-members:
```
Comment (joaander) [inline sphinx-doc/module-md-nlist.rst:18]:
```suggestion
    :no-inherited-members:
    :show-inheritance:
```
We are standardizing HOOMD docs to use `no-inerherited-members` and only document comment methods once in the base class. `show-inheritance` provides a link to that base class.

File: hoomd/md/nlist.py [hunk 9]
```diff
@@ -145,9 +152,10 @@ class Cell(_NList):
 
     Examples::
 
-        nl_c = nlist.cell(rebuild_check_delay = 1)
-        nl_c.buffer = 0.5
-        nl_c.exclusions = tuple();
+        nl_c = nlist.Cell(diameter_shift=True, nbuffer=0.5)
```
Comment (joaander) [inline hoomd/md/nlist.py:155]:
```suggestion
        nl_c = nlist.Cell(diameter_shift=True, buffer=0.5)
```

File: hoomd/md/nlist.py [hunk 10]
```diff
@@ -119,10 +124,12 @@ class Cell(_NList):
 
     Args:
         buffer (float):  Buffer width.
-        exclusions (tuple): ...
+        exclusions (tuple): Excludes pairs from the neighbor list, which
```
Comment (joaander) [inline hoomd/md/nlist.py:127]:
```suggestion
        exclusions (Tuple[str]): Excludes pairs from the neighbor list, which 
```
We are using explicit type annotations throughout the docs.

File: hoomd/md/nlist.py [hunk 11]
```diff
@@ -119,10 +124,12 @@ class Cell(_NList):
 
     Args:
```
Comment (joaander) [inline hoomd/md/nlist.py:125]: Document all of these arguments, except for `deterministic` under an `Attributes:` section of the base class `_Nlist` (you can copy and paste). `deterministic` should be documented under an `Attributes:` section of `Cell` at the end of the class docstring.

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks! Sorry for the delay on this review.

I have some change requests below. Also, could you revise the docs for `shortest_rebuild` - it doesn't need to be so wordy and should not reference old APIs. Maybe something like:
```
"""The shortest period between neighbor list rebuilds.

        `shortest_rebuild` is the smallest number of time steps between neighbor list rebuilds during the previous `Simulation.run`.
"""
```
(oh, and remove the -1 in the implementation)

The rest of the details there are more appropriate for a tutorial on tuning neighbor list performance which we can write at a later time.

I'd like to get this wrapped up and merged next week. Let me know if you don't think you can get it done by then.

File: hoomd/md/nlist.py [hunk 12]
```diff
@@ -53,8 +53,13 @@
 
 
 class nlist:
+    R""" 
+
+    Generates a class for other classes that have not been ported.
```
Comment (b-butler) [inline hoomd/md/nlist.py:58]:
```suggestion
    Obsolete parent class for neighbor list objects.
```

File: hoomd/md/nlist.py (hunk 10 above)
Comment (b-butler) [inline hoomd/md/nlist.py:127]: We 
```suggestion
        exclusions (tuple[str]): Excludes pairs from the neighbor list, which 
```
In an offline conversation, @joaander and I have decided that we will keep the `[]` notation, but switch to using the standard Python type names.

**Review by b-butler — CHANGES_REQUESTED** [review]
I have just a few more comments to address.

File: sphinx-doc/module-md-nlist.rst [hunk 13]
```diff
@@ -9,12 +9,11 @@ md.nlist
...
     md.nlist.Cell
-    md.nlist.stencil
-    md.nlist.tree
 
 .. rubric:: Details
 
 .. automodule:: hoomd.md.nlist
     :synopsis: Neighbor list acceleration structures.
-    :members:
+    :members: _Nlist, Cell
```
Comment (joaander) [inline sphinx-doc/module-md-nlist.rst:17]:
```suggestion
    :members: _NList, Cell
```

**Comment by joaander** [comment]
I've revised the docs for `Cell` and `_Nlist`.

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**